docs/DxcOnUnix.rst

```diff
@@ -0,0 +1,133 @@
+======================================
+DirectXShaderCompiler on Linux & macOS
+======================================
+
+.. contents::
+   :local:
+   :depth: 3
+
+Introduction
+============
+
+DirectXShaderCompiler (DXC) is based on LLVM/Clang, which is originally
+cross-platform. However, to support HLSL, certain Windows specific techniques
+(like COM, SAL, etc.) are introduced to solve technical issues on the Windows
+platform, which also makes DXC not compilable/runnable on non-Windows platforms.
+
+Upon `several <https://github.com/Microsoft/DirectXShaderCompiler/issues/1082>`_
+requests from the community, we have started the effort to enable compilation
+and running of DirectXShaderCompiler on non-Windows platforms (Linux and macOS).
+
+Current Status
+==============
+
+Up and Running
+--------------
+We have currently reached the point where we can successfully build and run DXC
+on Linux and macOS. We are also able to run SPIR-V CodeGen tests on these
+platforms.
+
+Known Limitations
+-----------------
+
+The following targets are currently disabled for non-Windows platforms and this
+is an area where further contributions can be made:
+
+* d3dcomp
+* dxa
+* dxopt
+* dxl
+* dxr
+* dxv
+* dxlib-sample
+
+Moreover, since the HLSL CodeGen tests were originally written with Windows in
+mind, they require the Windows-specific `TAEF Framework <https://docs.microsoft.com/en-us/windows-hardware/drivers/taef/>`_
+to run. Therefore we are not able to compile/run these tests on non-Windows
+platforms.
+
+Known Issues
+------------
+Running the SPIR-V CodeGen tests results in opening a large number of file
+descriptors, and if the OS limitation on the number of FDs allowed to be opened
+by a process is low, it will cause test failures. We have not seen this as an
+issue on Windows and Linux. On macOS we currently increase the allowed limit to
+get around the problem for the time being.
+
+Build Requirements
```

Comment:
Maybe we can fold all the following sections into one section "Building and Using" and make them subsections. They are all related to building in a sense.

Reply:
Done